integration/utils/contLoadGenerator.go

```diff
@@ -411,3 +403,186 @@ func (lg *ContLoadGenerator) sendTx(workerID int) {
+				if !stopped {
+					stopped = true
+					wg.Done()
+				}
+			}, // on expired
+			func(_ flowsdk.Identifier) {
+				lg.log.Warn().Str("tx_id", addKeysTx.ID().String()).Msgf("tx timed out")
+				if !stopped {
+					stopped = true
+					wg.Done()
+				}
+			}, // on timout
+			func(_ flowsdk.Identifier, err error) {
+				lg.log.Error().Err(err).Str("tx_id", addKeysTx.ID().String()).Msgf("tx error")
+				if !stopped {
+					stopped = true
+					wg.Done()
+				}
+			}, // on error
+			60)
+		wg.Add(1)
+		wg.Wait()
+	}
+}
+
+func (lg *ContLoadGenerator) sendCompHeavyTx(workerID int) {
+	// TODO move this as a configurable parameter
+	blockRef, err := lg.blockRef.Get()
+	if err != nil {
+		lg.log.Error().Err(err).Msgf("error getting reference block")
+		return
+	}
+
+	lg.log.Trace().Msgf("getting next available account")
+
+	acc := <-lg.availableAccounts
+	defer func() { lg.availableAccounts <- acc }()
+
+	txScript := lg.scriptCreator.CreateCompHeavyTransaction(30)
+	lg.log.Trace().Msgf("creating transaction")
+	tx := flowsdk.NewTransaction().
+		SetReferenceBlockID(blockRef).
+		SetScript(txScript).
+		SetProposalKey(*acc.address, 0, acc.seqNumber).
+		SetPayer(*acc.address).
+		AddAuthorizer(*acc.address)
+
+	lg.log.Trace().Msgf("signing transaction")
+	acc.signerLock.Lock()
+	err = tx.SignEnvelope(*acc.address, 0, acc.signer)
+	if err != nil {
+		acc.signerLock.Unlock()
+		lg.log.Error().Err(err).Msgf("error signing transaction")
+		return
+	}
+	acc.seqNumber++
+	acc.signerLock.Unlock()
+
+	lg.log.Trace().Msgf("sending transaction")
+	err = lg.flowClient.SendTransaction(context.Background(), *tx)
+	if err != nil {
+		lg.log.Error().Err(err).Msgf("error sending transaction")
+		return
+	}
+
+	lg.log.Trace().Msgf("tracking sent transaction")
+	lg.workerStatsTracker.AddTxSent()
+	lg.loaderMetrics.TransactionSent()
+
+	if lg.trackTxs {
+		stopped := false
+		wg := sync.WaitGroup{}
+		lg.txTracker.AddTx(tx.ID(),
+			nil,
+			func(_ flowsdk.Identifier, res *flowsdk.TransactionResult) {
+				lg.log.Trace().Str("tx_id", tx.ID().String()).Msgf("finalized tx")
+				if !stopped {
+					stopped = true
+					wg.Done()
+				}
+			}, // on finalized
+			func(_ flowsdk.Identifier, _ *flowsdk.TransactionResult) {
+				lg.log.Trace().Str("tx_id", tx.ID().String()).Msgf("sealed tx")
+			}, // on sealed
+			func(_ flowsdk.Identifier) {
+				lg.log.Warn().Str("tx_id", tx.ID().String()).Msgf("tx expired")
+				if !stopped {
+					stopped = true
+					wg.Done()
+				}
+			}, // on expired
+			func(_ flowsdk.Identifier) {
+				lg.log.Warn().Str("tx_id", tx.ID().String()).Msgf("tx timed out")
+				if !stopped {
+					stopped = true
+					wg.Done()
+				}
+			}, // on timout
+			func(_ flowsdk.Identifier, err error) {
+				lg.log.Error().Err(err).Str("tx_id", tx.ID().String()).Msgf("tx error")
+				if !stopped {
+					stopped = true
+					wg.Done()
+				}
+			}, // on error
+			60)
+		wg.Add(1)
+		wg.Wait()
+	}
+}
```

Comment:
I think this is identical across both transaction types. Can we refactor to avoid the duplication?